pkg/resource/supplier.go

```diff
@@ -1,8 +1,9 @@
 package resource
 
-// Resource Supplier supply the list of resource.Resource, its the front to retrieve remote resources
+// Supplier supply the list of resource.Resource, it's the main interface to retrieve state or remote resources
 type Supplier interface {
 	Resources() ([]*Resource, error)
+	SourceCount() int
```

Comment:
I don't think this make sense to add this to this interface. For remote ones we should be able to retrieve resources without implementing any `SourceCount` method.

Maybe we should introduce a new interface like `IacSupplier` and compose the `TerraformStateReader` with both `Supplier` and `IacSupplier` WDYT ?